Export DC to mkversion.sh run in shell function

nemanja-boric-sociomantic opened the pull request:
Shell function doesn't inherit variables exported
in Makefile, so explicit passing of `DC` is performed.

Fixes #23

nemanja-boric-sociomantic committed:
Export DC to mkversion.sh run in shell function

Shell function doesn't inherit variables exported
in Makefile, so explicit passing of `DC` is performed.

Issue describing this behaviour of make:
https://bugs.debian.org/184864

Fixes #23

leandro-lucarella-sociomantic reviewed (changes requested):
`DC` is supposed to be exported, see https://github.com/sociomantic-tsunami/makd/blob/da527f1e33e00f1a0d02745d0fe366e602639c83/Makd.mak#L106-L108. My guess is `?=` is what is making it fail on some circumstances, I guess something like:

``` make
DC ?= ...
export DC
```

Should fix it.

nemanja-boric-sociomantic commented:
@leandro-lucarella-sociomantic I'm not sure that `DC` is exported to the subshell when called with `$(shell`

nemanja-boric-sociomantic commented:
https://bugs.debian.org/184864

nemanja-boric-sociomantic commented:
But, I'll try with `export DC`

nemanja-boric-sociomantic commented:
Yeah, it doesn't work.

leandro-lucarella-sociomantic commented:
:cry:

leandro-lucarella-sociomantic commented:
Add a comment with a link to the issue maybe?

nemanja-boric-sociomantic commented:
Updated with the issue link in the commit message.

mathias-lang-sociomantic commented:
I'd also make the default in `mkversion.sh` refers to `dmd` instead of DMD1.
It shouldn't affect us, but will make the script more usable for upstream.

leandro-lucarella-sociomantic reviewed (changes requested):
> I'd also make the default in mkversion.sh refers to dmd instead of DMD1.
> It shouldn't affect us, but will make the script more usable for upstream.

Yeah, I haven't saw the original issue before, it definitely make sense to change the default too (as a separate commit because they are 2 different things, but I think both are needed to fix the original issue properly).

nemanja-boric-sociomantic committed:
Default to dmd instead of dmd1 if DC is not found

If DC environment variable is not set, mkversion.sh
will now default to dmd compiler, instead of dmd1.

nemanja-boric-sociomantic commented:
Updated.

mathias-lang-sociomantic reviewed (approved):
LGTM, thanks :)

leandro-lucarella-sociomantic commented:
> Updated with the issue link in the commit message.

Mmm, I meant a comment in the code (too), as I have the feeling someone (like me) would see explicit env var passing in the future and think is redundant and remove it :)

nemanja-boric-sociomantic commented:
You're such a pain :-).

Updated.

leandro-lucarella-sociomantic commented:
> You're such a pain :-).

:kissing_heart:
